pkg/apis/core/v1/helper/helpers.go

```diff
@@ -127,19 +83,14 @@ func HugePageSizeFromMedium(medium v1.StorageMedium) (resource.Quantity, error)
 // IsOvercommitAllowed returns true if the resource is in the default
 // namespace and is not hugepages.
 func IsOvercommitAllowed(name v1.ResourceName) bool {
-	return IsNativeResource(name) &&
-		!IsHugePageResourceName(name)
-}
-
-// IsAttachableVolumeResourceName returns true when the resource name is prefixed in attachable volume
-func IsAttachableVolumeResourceName(name v1.ResourceName) bool {
-	return strings.HasPrefix(string(name), v1.ResourceAttachableVolumesPrefix)
+	return corev1helpers.IsNativeResource(name) &&
+		!corev1helpers.IsHugePageResourceName(name)
 }
 
 // IsScalarResourceName validates the resource for Extended, Hugepages, Native and AttachableVolume resources
 func IsScalarResourceName(name v1.ResourceName) bool {
-	return IsExtendedResourceName(name) || IsHugePageResourceName(name) ||
-		IsPrefixedNativeResource(name) || IsAttachableVolumeResourceName(name)
+	return corev1helpers.IsExtendedResourceName(name) || corev1helpers.IsHugePageResourceName(name) ||
+		corev1helpers.IsPrefixedNativeResource(name) || corev1helpers.IsAttachableVolumeResourceName(name)
```

Comment:
Is this a bug? shouldn't this call `IsNativeResource`?

Reply:
Based on the usage of `IsScalarResourceName` under `pkg/scheduler` it is not. `IsScalarResourceName` is always called inside a switch

```
switch resource {
	case v1.ResourceCPU:
		return nodeInfo.Allocatable.MilliCPU, (nodeInfo.NonZeroRequested.MilliCPU + podRequest)
	case v1.ResourceMemory:
		return nodeInfo.Allocatable.Memory, (nodeInfo.NonZeroRequested.Memory + podRequest)

	case v1.ResourceEphemeralStorage:
		return nodeInfo.Allocatable.EphemeralStorage, (nodeInfo.Requested.EphemeralStorage + podRequest)
	default:
		if v1helper.IsScalarResourceName(resource) {
			return nodeInfo.Allocatable.ScalarResources[resource], (nodeInfo.Requested.ScalarResources[resource] + podRequest)
		}
	}
```

which first checks for resources (not containing `kubernetes.io/` string). Calling `IsScalarResourceName` with `IsNativeResource` would be redundant (unless there are other relevent resources to catch).

Also, I plan to move `IsScalarResourceName` under the `pkg/scheduler` which requires `IsPrefixedNativeResource` to be exported.

Reply:
Sorry to continue move this conversation to a tangent, but, do we account for `cpu` and `kubernetes.io/cpu` differently? Or is this something that gets normalized in apiserver?

Reply:
I don't see any line of code suggesting there's any normalization anywhere. You are most likely allowed to set `kubernetes.io/cpu` in your resource requests but the resource will not get recognized as `cpu` (at least by the scheduler).

I don't know if there's any validation forbidding to set `kubernetes.io/cpu`.

@derekwaynecarr do you know who might now answer to @alculquicondor 's question?

Reply:
ping @derekwaynecarr